src/analysis/retail/evoker/preservation/modules/talents/TimeOfNeed.tsx

```diff
@@ -0,0 +1,241 @@
+          if (eventList.events.length !== 0) {
+            //Iterate all the damage taken events during the time window
+            eventList.events.forEach((hit) => {
+              hit = hit as DamageEvent;
+              //Check if the hit landed on the same player that the Time of Need targeted
+              if (hit.targetID === spawn.target) {
+                //If the hitpoints after the hit are equal or less than 0, the player died anyway
+                if (hit.hitPoints !== undefined && hit.hitPoints === 0) {
+                  this.spawns[index].result = Result.Dead;
+                } else {
+                  //Initialize the healing from Time of Need with the VE
+                  let tonHealingAtThisPoint = spawn.verdantEmbrace?.amount || 0;
+                  //If there is no living flames casted, or this hit was before the first living flame landed
+                  if (
+                    spawn.livingFlames.length === 0 ||
+                    hit.timestamp < spawn.livingFlames[0].timestamp
+                  ) {
+                    //Check if the player would've died from this hit without the Verdant Embrace healing
+                    if (
+                      hit.hitPoints &&
+                      hit.hitPoints > 0 &&
+                      hit.hitPoints - tonHealingAtThisPoint <= 0
+                    ) {
+                      this.spawns[index].result = Result.Proc; //Person was saved by the VE
+                    }
+                  } else {
+                    //Add up all the healing done by the living flames that happened before this hit
+                    spawn.livingFlames.forEach((livingFlame) => {
+                      if (livingFlame.timestamp < hit.timestamp) {
+                        tonHealingAtThisPoint += livingFlame.amount;
+                      }
+                    });
+                    //If the hit points after this hit minus the total ToN healing done to this point is equal or less than 0, then he was saved
+                    if (
+                      hit.hitPoints &&
+                      hit.hitPoints > 0 &&
+                      hit.hitPoints - tonHealingAtThisPoint <= 0
+                    ) {
+                      //Mark as long save
+                      this.spawns[index].result = Result.Long;
+                    }
+                  }
+                }
+              }
+            });
+          }
+        } else {
+          //If ToN did no healing, it bugged out
+          this.spawns[index].result = Result.Bug;
+        }
+      }
+    });
+  }
+
+  async load() {
+    //Run every ToN spawn through parseDamageTaken()
+    for (const spawn of this.spawns) {
+      await fetchWcl(`report/events/damage-taken/${this.owner.report.code}`, {
+        start: spawn.summon.timestamp,
+        end: spawn.summon.timestamp + 8000,
+      }).then((json) => {
```

Comment:
rather than `await`ing a `then`ed promise, just await the `fetchWcl` directly and use the type parameter:

```ts
const json = await fetchWcl<WCLEventsResponse>(...);
```